torch_frame/data/multi_nested_tensor.py

```diff
@@ -78,85 +78,161 @@ def from_tensor_mat(
     def __repr__(self) -> str:
         name = ' '.join([
             f"{self.__class__.__name__}(num_rows={self.num_rows},",
-            f"num_cols={self.num_cols},", f"device={self.values.device})"
+            f"num_cols={self.num_cols},", f"device={self.device})"
         ])
-
         return name
 
     def __getitem__(
```

Comment:
This method here is super complex. Can we refactor it?

Reply:
Yes, will try.